src/imputation/MoR.py

```diff
@@ -154,22 +154,32 @@ def calculate_growth_rates(current_df, prev_df, target_vars):
     current_df = current_df.copy().loc[current_df["formtype"] == "0001", :]
     # prev_df = prev_df.copy().loc[prev_df["formtype"] == "0001", :]
 
+    # Ensure we only have one row per reference/imp_class for previous and current data
+    prev_df = (
+        prev_df[["reference", "imp_class"] + target_vars]
+        .groupby(["reference", "imp_class"])
+        .sum()
+    ).reset_index()
+
+    current_df = (
+        current_df[["reference", "imp_class"] + target_vars]
+        .groupby(["reference", "imp_class"])
+        .sum()
+    ).reset_index()
+
+    # Merge the clear current and previous data
```

Comment:
looks a bit easier to understand.

Do we need to do "drop=True" in reset index? I know sometimes you lose a col.

Reply:
Not here I don't think - the index will be ["reference", "imp_class"] and we don't want to lose these.